app/src/test/kotlin/com/softteco/template/ui/feature/profile/ProfileViewModelTest.kt

```diff
@@ -0,0 +1,85 @@
+package com.softteco.template.ui.feature.profile
+
+import androidx.lifecycle.viewModelScope
+import com.softteco.template.data.base.error.ErrorEntity
+import com.softteco.template.data.base.error.Result
+import com.softteco.template.data.profile.ProfileRepository
+import com.softteco.template.data.profile.entity.Profile
+import com.softteco.template.utils.MainDispatcherExtension
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import java.util.UUID
+import kotlin.time.Duration.Companion.seconds
+
+@ExtendWith(MainDispatcherExtension::class)
+class ProfileViewModelTest {
+
+    @Test
+    fun `when screen is open and data isn't received then loading is shown`() {
+        // Arrange
+        val profile = Profile(id = UUID.randomUUID().toString(), name = "John Doe")
+        val profileRepository = object : ProfileRepository {
```

Comment:
Let's mock a repository, it should simplify tests.  We could use Mockk lib for that.